include/bluetooth/enocean.h

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @defgroup bt_enocean Bluetooth EnOcean library API
+ * @{
+ * @brief API for interacting with EnOcean Bluetooth wall switches and sensors
+ */
+
+#ifndef BT_ENOCEAN_H__
+#define BT_ENOCEAN_H__
+
+#include <bluetooth/bluetooth.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Double rocker switch top left button (OA) */
+#define BT_ENOCEAN_SWITCH_OA BIT(0)
+/** Double rocker switch bottom left button (IA) */
+#define BT_ENOCEAN_SWITCH_IA BIT(1)
+/** Double rocker switch top right button (OB) */
+#define BT_ENOCEAN_SWITCH_OB BIT(2)
+/** Double rocker switch bottom right button (IB) */
+#define BT_ENOCEAN_SWITCH_IB BIT(3)
+
+/** Single rocker switch top button (O) */
+#define BT_ENOCEAN_SWITCH_O BT_ENOCEAN_SWITCH_OB
+/** Single rocker switch bottom button (I) */
+#define BT_ENOCEAN_SWITCH_I BT_ENOCEAN_SWITCH_IB
+
+/** EnOcean sensor data message. */
+struct bt_enocean_sensor_data {
+	bool *occupancy; /**< Current occupancy status. */
+	u16_t *light_sensor; /**< Light level in sensor (lx). */
+	u16_t *battery_voltage; /**< Backup battery voltage (mV). */
+	u16_t *light_solar_cell; /**< Light level in solar cell (lx). */
+	u8_t *energy_level; /**< Energy level (percent). */
+};
+
+/** EnOcean device representation. */
+struct bt_enocean_device {
+	u32_t seq;  /**< Most recent sequence number. */
+	u8_t flags; /**< Internal flags field. */
+	s8_t rssi; /**< Most recent RSSI measurement. */
+	u8_t addr[6]; /**< Device address. */
```

Comment:
Address without address type? Should it always be intepreted as a random address? Or public?
Could we use either bt_addr_le_t or bt_addr_t for this field?

Reply:
Good point. These are always random addresses, but should probably be changed for correctness.